sdk-actors/src/main/java/io/dapr/actors/client/ActorProxyImpl.java

```diff
@@ -136,29 +136,33 @@ public Object invoke(Object proxy, Method method, Object[] args) {
       throw new UnsupportedOperationException("Actor methods can only have zero or one arguments.");
     }
 
+    ActorMethod actorMethodAnnotation = method.getDeclaredAnnotation(ActorMethod.class);
+    String methodName = method.getName();
+    if ((actorMethodAnnotation != null) && !actorMethodAnnotation.name().isEmpty()) {
+      methodName = actorMethodAnnotation.name();
+    }
+
     if (method.getParameterCount() == 0) {
       if (method.getReturnType().equals(Mono.class)) {
-        ActorMethod actorMethodAnnotation = method.getDeclaredAnnotation(ActorMethod.class);
         if (actorMethodAnnotation == null) {
-          return invokeMethod(method.getName());
+          return invokeMethod(methodName);
         }
 
-        return invokeMethod(method.getName(), actorMethodAnnotation.returns());
+        return invokeMethod(methodName, actorMethodAnnotation.returns());
       }
 
-      return invokeMethod(method.getName(), method.getReturnType()).block();
+      return invokeMethod(methodName, method.getReturnType()).block();
     }
 
     if (method.getReturnType().equals(Mono.class)) {
-      ActorMethod actorMethodAnnotation = method.getDeclaredAnnotation(ActorMethod.class);
       if (actorMethodAnnotation == null) {
-        return invokeMethod(method.getName(), args[0]);
+        return invokeMethod(methodName, args[0]);
       }
 
-      return invokeMethod(method.getName(), args[0], actorMethodAnnotation.returns());
+      return invokeMethod(methodName, args[0], actorMethodAnnotation.returns());
```

Comment:
Do we need any special case logic for the case for `Void.class` as the value of `returns()` the old behavior (not specifying return type for `Mono`) would send it down the code path at line 159

Reply:
Void is a valid option. So we don't know if customers intended Void or if they just forgot and left the default there.

Reply:
Updated with logic now that we have the "Undefined" class.